python/rapidsmpf/rapidsmpf/memory/buffer_resource.pyx

```diff
@@ -188,11 +196,15 @@ cdef class BufferResource:
         # TODO: once RMM is migrating to CCCL (copyable) any_resource,
         # rather than the any_resource_ref reference type, we don't
         # need to keep this alive here.
-        self._mr = device_mr
+        self._device_mr = device_mr
+        self._pinned_mr = pinned_mr
+        cdef shared_ptr[cpp_PinnedMemoryResource] _pinned_mr
+        if self._pinned_mr is not None:
+            _pinned_mr = self._pinned_mr._handle
```

Comment:
Do we need to `self._pinned_mr.reset()` in `__dealloc__`?

Reply:
Right, but there only its own reference is reset, the reference here may still remain. I think as is `__dealloc__` will call `self._handle.reset()` which will not guarantee `self._pinned_mr` will be reset until any potential cyclic references are broken and the garbage collector runs, so to be safe I think we should call `self._pinned_mr.reset()` as well, otherwise this reference may still outlive `PinnedMemoryResource`.